File: csharp/acronym/Acronym.cs
using System;
using System.Linq;

public static class Acronym
{
    public static string Abbreviate(string phrase)
    {
        char[] delimiters = new[] { ' ', '-', '_' };
        return phrase.Split(delimiters, StringSplitOptions.RemoveEmptyEntries).Select(word => Char.ToUpper(word[0]).ToString()).Aggregate( (w, v) => w + v );
    }
}

File: fsharp/acronym/Acronym.fs
module Acronym

open System

let abbreviate (phrase: string) = 
    phrase.Split([|' '; '-'; '_'|], StringSplitOptions.RemoveEmptyEntries)
    |> Seq.map (fun word -> word.[0] |> Char.ToUpper |> string)
    |> Seq.fold (fun acc c -> acc + c) ""